File: PCIC/2023/test/pC/py/grader.py
def is_broken(k: int) -> int:
    if not type(k) is int:
        wa("invalid broken query")
    black_box.count += 1
    if black_box.count > 60:
        wa("too many queries")
    if k <= 0 or k > black_box.M:
        wa("invalid broken query")
    if k > black_box.h:
        return 1
    return 0

if __name__ == '__main__':
    M, h = map(int, input().split(' '))

    import Egg

    def wa(msg):
        print('Wrong Answer: ' + msg)
        exit(0)

    class Black_Box:
        def __init__(self, M, h):
            self.count = 0
            self.M = M
            self.h = h

    black_box = Black_Box(M, h)

    rt = Egg.height_limit(M)
    if rt != h:
        wa("incorrect height")
    else:
        print("Accepted: " + str(black_box.count))

File: PCIC/2023/test/pC/cpp/grader.cpp
#include "Egg.h"

#include <cassert>
#include <stdio.h>
#include <stdlib.h>

namespace {
    long long M, h;
    int count;
    void wa(const char* msg) {
        printf("Wrong Answer: %s\n", msg);
        exit(0);
    }
}

int is_broken(long long k) {
    ++count;
    if (count > 60)
        wa("too many queries");
    if (k <= 0 || k > M)
        wa("invalid broken query");
    return k > h;
}

int main() {
    assert(2 == scanf("%lld %lld", &M, &h));
    long long rt = height_limit(M);
    if (rt == h)
        printf("Accepted: %d\n", count);
    else
        wa("incorrect height");
}
